metpy/calc/cross_sections.py

```diff
@@ -0,0 +1,307 @@
+# Copyright (c) 2017 MetPy Developers.
+# Distributed under the terms of the BSD 3-Clause License.
+# SPDX-License-Identifier: BSD-3-Clause
+"""Contains calculation of various derived indices.
+
+Compared to the rest of the calculations which are based around pint quantities, this module
+is based around xarray DataArrays.
+"""
+
+import cartopy.crs as ccrs
+import numpy as np
+import xarray as xr
+
+from .basic import coriolis_parameter
+from .tools import first_derivative
+from ..package_tools import Exporter
+from ..xarray import CFConventionHandler, check_matching_coordinates
+
+exporter = Exporter(globals())
+
+
+def distances_from_cross_section(cross):
+    """Calculate the distances in the x and y directions along a cross-section.
+
+    Parameters
+    ----------
+    cross : `xarray.DataArray`
+        The input DataArray of a cross-section from which to obtain geometeric distances in
+        the x and y directions.
+
+    Returns
+    -------
+    x, y : tuple of `xarray.DataArray`
+        A tuple of the x and y distances as DataArrays
+
+    """
+    if (CFConventionHandler.check_axis(cross.metpy.x, 'lon') and
+            CFConventionHandler.check_axis(cross.metpy.y, 'lat')):
+        # Use pyproj to obtain x and y distances
+        from pyproj import Geod
+
+        g = Geod(cross.metpy.cartopy_crs.proj4_init)
+        lon = cross.metpy.x
+        lat = cross.metpy.y
+
+        forward_az, _, distance = g.inv(lon[0].values * np.ones_like(lon),
```

Comment:
Is this assuming 2D lon/lat?

Reply:
No, I wrote it assuming 1D lon/lat (as coordinate variables along the cross-section index dimension), in order to return 1D x/y (again, along cross-section index).